src/x_connective.c

```diff
@@ -1568,9 +1576,9 @@ void value_release(t_symbol *s)
 int value_getfloat(t_symbol *s, t_float *f)
 {
     t_vcommon *c = (t_vcommon *)pd_findbyclass(s, vcommon_class);
-    if (!c)
+    if (!c || c->c_atom.a_type != A_FLOAT)
```

Comment:
the question is if we should return false for a type mismatch or just return a default value... originally, these methods only returned false if the value object didn't exist.

Reply:
This problem manifests itself in [expr] as mentioned by @porres in https://github.com/pure-data/pure-data/pull/817#issuecomment-562874271

Reply:
It now prints a different error: "var 'foo' is not a float"